### PR TITLE
[node-manager][docs] Improved NodeGroupConfiguration documentation

### PR DESCRIPTION
**Pull request by Zhbert**
## Description

Added additional information about using the content parameter in NodeGroupConfiguration.

## Why do we need it, and what problem does it solve?

This will improve the understanding of the documentation.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [x] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries
 
```changes
section: docs
type: chore
summary: Minor documentation updates of the node-manager module.
impact_level: low
```

### CODE REVIEW

File: modules/040-node-manager/crds/nodegroupconfiguration.yaml [hunk 1]
```diff
@@ -34,7 +34,24 @@ spec:
...
+                    Содержимое шага конфигурации в виде bash-скрипта.
+
+                    Для генерации скрипта возможно использовать [Go Template](https://pkg.go.dev/text/template).
+
+                    Получить список доступных для использования в шаблонах параметров можно из Secret'а `bashible-apiserver-context`:
+                    ```
+                    kubectl -n d8-cloud-instance-manager get secrets bashible-apiserver-context -o yaml | grep -w 'input.yaml:' | awk '{ print $2}' | base64 --decode
+                    ```
+
+                    Также в скрипте допустимо использовать подготовленные команды [bashbooster](https://github.com/deckhouse/deckhouse/tree/main/candi/bashible/bashbooster). Например:
```
Comment (kirkonru) [inline modules/040-node-manager/crds/nodegroupconfiguration.yaml:47]:
```suggestion
                    A bash script that does the same things you would do in a configuration step.

                    You can use [Go Template](https://pkg.go.dev/text/template) to generate a script.

                    You can get the list of parameters available for use in templates from the `bashible-apiserver-context` Secret:
                    ```
                    kubectl -n d8-cloud-instance-manager get secrets bashible-apiserver-context -o yaml | grep -w 'input.yaml:' | awk '{ print $2}' | base64 --decode
                    ```

                    You can also use the pre-defined [bashbooster](https://github.com/deckhouse/deckhouse/tree/main/candi/bashible/bashbooster) commands in the script. For example:
```

File: modules/040-node-manager/crds/doc-ru-nodegroupconfiguration.yaml [hunk 2]
```diff
@@ -23,7 +23,24 @@ spec:
             spec:
               properties:
                 content:
-                  description: Содержимое шага конфигурации. Может быть или шаблоном Go Template, или bash-скриптом.
+                  description: |
+                    Содержимое шага конфигурации в виде bash-скрипта.
+
+                    Для генерации скрипта возможно использовать [Go Template](https://pkg.go.dev/text/template).
+
+                    Получить список доступных для использования в шаблонах параметров можно из Secret'а `bashible-apiserver-context`:
```
Comment (z9r5) [inline modules/040-node-manager/crds/doc-ru-nodegroupconfiguration.yaml:31]:
```suggestion
                    Получить список параметров, доступных для использования в шаблонах можно из секрета `bashible-apiserver-context`:
```

File: modules/040-node-manager/docs/README.md [hunk 3]
```diff
@@ -103,6 +103,118 @@ There are two ways for setting the number of nodes in a group when nodes are pro
...
+- обновления ядра ОС на конкретную версию ([пример](faq.html#как-обновить-ядро-на-узлах));
+- изменение параметров ОС ([пример настройки параметра sysctl](examples.html#задание-параметра-sysctl));
+- сбор информации на узле и выполнение других подобных действий. 
+
+Ресурс `NodeGroupConfiguration` позволяет указывать [приоритет](cr.html#nodegroupconfiguration-v1alpha1-spec-weight) выполняемым скриптам, ограничивать их выполнение определенными [группами узлов](cr.html#nodegroupconfiguration-v1alpha1-spec-nodegroups) и [типами ОС](cr.html#nodegroupconfiguration-v1alpha1-spec-bundles).
+
+Код скрипта указывается в параметре [content](cr.html#nodegroupconfiguration-v1alpha1-spec-content) ресурса. При создании скрипта на узле содержимое параметра `content` проходит через шаблонизатор [Go Template](https://pkg.go.dev/text/template), который позволят встроить дополнительный уровень логики при генерации скрипта. При прохождении через шаблонизатор становится доступным контекст с набором динамических переменных.
+
+Переменные, которые доступны для использования в шаблонизаторе:
```
Comment (kirkonru) [inline modules/040-node-manager/docs/README.md:118]:
```suggestion
The following variables are supported by the templating engine: 
```

File: modules/040-node-manager/docs/README.md [hunk 4]
```diff
@@ -103,6 +103,118 @@ There are two ways for setting the number of nodes in a group when nodes are pro
...
+- сбор информации на узле и выполнение других подобных действий. 
+
+Ресурс `NodeGroupConfiguration` позволяет указывать [приоритет](cr.html#nodegroupconfiguration-v1alpha1-spec-weight) выполняемым скриптам, ограничивать их выполнение определенными [группами узлов](cr.html#nodegroupconfiguration-v1alpha1-spec-nodegroups) и [типами ОС](cr.html#nodegroupconfiguration-v1alpha1-spec-bundles).
+
+Код скрипта указывается в параметре [content](cr.html#nodegroupconfiguration-v1alpha1-spec-content) ресурса. При создании скрипта на узле содержимое параметра `content` проходит через шаблонизатор [Go Template](https://pkg.go.dev/text/template), который позволят встроить дополнительный уровень логики при генерации скрипта. При прохождении через шаблонизатор становится доступным контекст с набором динамических переменных.
+
+Переменные, которые доступны для использования в шаблонизаторе: 
+<ul>
+<li><code>.cloudProvider</code> (для групп узлов с nodeType <code>CloudEphemeral</code> или <code>CloudPermanent</code>) — массив данных облачного провайдера.
+{% offtopic title="Пример данных..." %}
```
Comment (kirkonru) [inline modules/040-node-manager/docs/README.md:121]:
```suggestion
<li><code>.cloudProvider</code> (for node groups of nodeType <code>CloudEphemeral</code> or <code>CloudPermanent</code>) — cloud provider data.
{% offtopic title="Example of data..." %}
```

File: modules/040-node-manager/docs/README.md [hunk 5]
```diff
@@ -103,6 +103,118 @@ There are two ways for setting the number of nodes in a group when nodes are pro
...
+  region: region2
+  type: openstack
+  zones:
+  - nova
+```
+{% endofftopic %}</li>
+<li><code>.cri</code> — используемый CRI (с версии Deckhouse 1.49 используется только <code>Containerd</code>).</li>
+<li><code>.kubernetesVersion</code> — используемая версия Kubernetes.</li>
+<li><code>.nodeUsers</code> — массив данных группы узлов.
+{% offtopic title="Пример данных..." %}
```
Comment (kirkonru) [inline modules/040-node-manager/docs/README.md:154]:
```suggestion
<li><code>.cri</code> — the CRI in use (starting with Deckhouse 1.49, only <code>Containerd</code> is supported).</li>
<li><code>.kubernetesVersion</code> — the Kubernetes version in use.</li>
<li><code>.nodeUsers</code> — node group data.
{% offtopic title="Example of data..." %}
```

Comment (kirkonru) [reply modules/040-node-manager/docs/README.md:154]: Got to be **_node user_** data here, given that the node group data is below 🤔  By the way, what's **_массив_** in "массив данных"?

File: modules/040-node-manager/docs/README.md [hunk 6]
```diff
@@ -103,6 +103,118 @@ There are two ways for setting the number of nodes in a group when nodes are pro
...
+    nodeGroups:
+    - '*'
+    passwordHash: PASSWORD_HASH
+    sshPublicKey: SSH_PUBLIC_KEY
+    uid: 1050
+```
+{% endofftopic %}
+</li>
+<li><code>.nodeGroup</code> — массив данных группы узлов.
+{% offtopic title="Пример данных..." %}
```
Comment (kirkonru) [inline modules/040-node-manager/docs/README.md:169]:
```suggestion
<li><code>.nodeGroup</code> — node group data.
{% offtopic title="Example of data..." %}
```

**Review by kirkonru — APPROVED** [review]

**Pull request merged**